File: 120-triangle/120-triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        
        row_size = len(triangle)
        
        for i in range(row_size - 2, -1, -1):

            col_size = len(triangle[i])

            for j in range (col_size):
                
                triangle[i][j] += min(triangle[i+1][j], triangle[i+1][j+1])
                
        return triangle[0][0]

File: 120-triangle/120-triangle.cpp
# define MIN(a,b) (a<b?a:b)

class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        
        int row_size = triangle.size();
        int col_size;
        
        for(int i = row_size - 2; i >= 0; i--){
            
            col_size = triangle[i].size();
            
            for(int j = 0; j < col_size; j++){
                
                triangle[i][j] += MIN(triangle[i+1][j], triangle[i+1][j+1]);
            }
            
        }
        
        return triangle[0][0];
    }
};